### PR TITLE
tests: add ssh proxy to be able to collect logs from workers and other instances

### PR DESCRIPTION
**Pull request by cpanato**
fixes: INST-700

### CODE REVIEW

File: tests/rspec/lib/aws_cluster.rb [hunk 1]
```diff
@@ -66,6 +66,22 @@ def master_ip_address
     master_ip_addresses[0]
   end
 
+  def workers_ip_addresses
```
Comment (mxinden) [inline tests/rspec/lib/aws_cluster.rb]: To align with `master_ip_addresses`:

`s/workers_ip_addresses/worker_ip_addresses`

File: tests/rspec/lib/aws_cluster.rb [hunk 2]
```diff
@@ -66,6 +66,22 @@ def master_ip_address
...
+    ssh_workers_ip_addresses = []
+    Dir.chdir(@build_path) do
+      terraform_state = `terraform state show module.workers.aws_autoscaling_group.workers`.chomp.split("\n")
```
Comment (mxinden) [inline tests/rspec/lib/aws_cluster.rb]: Why not use https://github.com/coreos/tectonic-installer/blob/master/tests/rspec/lib/tfstate_file.rb#L9 here?

File: tests/rspec/lib/aws_cluster.rb (hunk 1 above)
Comment (cpanato) [reply tests/rspec/lib/aws_cluster.rb]: done

File: tests/rspec/lib/aws_cluster.rb (hunk 2 above)
Comment (cpanato) [reply tests/rspec/lib/aws_cluster.rb]: done

**Comment by mxinden** [comment]
@cpanato Let me know when I should have another look.

**Comment by cpanato** [comment]
@mxinden can you please do another round?

File: tests/rspec/lib/aws_cluster.rb [hunk 3]
```diff
@@ -48,24 +48,32 @@ def stop
...
-        instances_id = AwsSupport.sorted_auto_scaling_instances(attributes[1].strip.chomp, @aws_region)
-        instances_id.each do |instance_id|
-          ssh_master_ip_addresses.push AwsSupport.preferred_instance_ip_address(instance_id, @aws_region)
-        end
-      end
+
+    aws_autoscaling_group_master = @tfstate_file.value('module.masters.aws_autoscaling_group.masters', 'id')
+    instances_id = AwsSupport.sorted_auto_scaling_instances(aws_autoscaling_group_master, @aws_region)
+    instances_id.each do |instance_id|
+      ssh_master_ip_addresses.push AwsSupport.preferred_instance_ip_address(instance_id, @aws_region)
```
Comment (alexsomesan) [inline tests/rspec/lib/aws_cluster.rb]: Just as an observation, you can achieve this effect of constructing the `ssh_master_ip_addresses` array via the `.map` operation on `instances_id`, like this:

`instances_id.map { |instance_id| AwsSupport.preferred_instance_ip_address(instance_id, @aws_region) }

This ^ expression will evaluate to the same list of master IP addresses, but you would not have to use an explicit variable like `ssh_master_ip_addresses` for that. You could just use that expression as the return statement.

Not saying you should change anything, just wanted to let you about this Ruby feature.

Comment (cpanato) [reply tests/rspec/lib/aws_cluster.rb]: great! forgot about the map thing. thanks so much!

Comment (cpanato) [reply tests/rspec/lib/aws_cluster.rb]: done

File: tests/rspec/lib/aws_cluster.rb [hunk 4]
```diff
@@ -47,25 +47,23 @@ def stop
...
-        attributes = value.split('=')
-        next unless attributes[0].strip.eql?('id')
-        instances_id = AwsSupport.sorted_auto_scaling_instances(attributes[1].strip.chomp, @aws_region)
-        instances_id.each do |instance_id|
-          ssh_master_ip_addresses.push AwsSupport.preferred_instance_ip_address(instance_id, @aws_region)
-        end
-      end
-    end
-    ssh_master_ip_addresses
+    aws_autoscaling_group_master = @tfstate_file.value('module.masters.aws_autoscaling_group.masters', 'id')
```
Comment (mxinden) [inline tests/rspec/lib/aws_cluster.rb]: @cpanato Thanks for updating it here as well.

**Comment by cpanato** [comment]
retest this please

**Pull request merged**